app/src/main/java/de/xikolo/utils/extensions/IntentExtensions.kt

```diff
@@ -46,19 +53,90 @@ fun <T : Intent> T.createChooser(context: Context, title: String? = null, packag
     }
 }
 
+private fun <T : Intent> T.open(
+    activity: FragmentActivity,
+    forceChooser: Boolean,
+    parentFile: File? = null
+): Boolean {
+    val chooserIntent = createChooser(App.instance)
+
+    fun startActivity(): Boolean {
+        return try {
+            activity.startActivity(
+                if (forceChooser) {
+                    chooserIntent
+                } else {
+                    this
+                }
+                    ?.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+            )
+            true
+        } catch (e: Exception) {
+            activity.showToast(R.string.error_plain)
+            false
+        }
+    }
+
+    return if (chooserIntent == null) {
+        val dialog = UnsupportedIntentDialogAutoBundle.builder()
+            .apply {
+                if (parentFile != null) {
+                    fileMimeType(type)
+                }
+            }
+            .build()
+        dialog.listener = object : UnsupportedIntentDialog.Listener {
+            override fun onOpenPath() {
+                try {
+                    val fileManagerIntent = Intent(ACTION_GET_CONTENT)
+                    fileManagerIntent.setDataAndType(
+                        FileProviderUtil.getUriForFile(
+                            parentFile!!
+                        ),
+                        "vnd.android.document/directory"
+                    )
+                    fileManagerIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
+                    activity.startActivity(fileManagerIntent)
+                } catch (e: Exception) {
+                    activity.showToast(R.string.error_plain)
```

Comment:
Can we provide a more specific error message here? "No file manager found" or similar?